app/main/views/notifications.py

```diff
@@ -0,0 +1,39 @@
+# coding: utf-8
+from flask_login import current_user
+from flask import (
+    flash,
+    redirect,
+    render_template,
+    request,
+    url_for
+)
+from .. import main
+from ... import data_api_client
+
+
+@main.route('/notifications/user-research', methods=["GET", "POST"])
+def user_research_consent():
+    if current_user.is_anonymous():
+        return redirect(url_for('main.render_login', next=url_for('main.user_research_consent')))
+
+    if request.method == "POST":
+        userResearchOptIn = True if request.form.get('user_research_opt_in') == "True" else False
+        data_api_client.update_user(
+            current_user.id,
+            user_research_opted_in=userResearchOptIn,
+            updater=current_user.email_address
+        )
+
+        flash("Your preference has been saved", 'success')
+    else:
+        user = data_api_client.get_user(current_user.id)
+        userResearchOptIn = user['users']['userResearchOptedIn']
+
+    if current_user.role == 'supplier':
+        dashboard_url = url_for('external.supplier_dashboard')
+    else:
+        dashboard_url = url_for('external.buyer_dashboard')
+
+    return render_template("notifications/user-research-consent.html",
+                           userResearchOptIn=userResearchOptIn,
```

Comment:
This function call is formatted a bit weirdly.
Here you've aligned the kwargs with the opening delimiter, but further up in the call to `update_user` they're indented at 4 spaces.
```
my_function(foo,
                     bar,
                      baz)
```
```
my_function(
    foo,
    bar,
    baz
)
```
It was probably your editor, mine does that too sometimes. I personally prefer the second way. It keeps spacing standard at multiples of 4. Whereas in this function call the kwargs are indented to 27 spaces.

Regardless, it's best to pick one and stick with it.